### DISCUSSION TITLE
CSS value functions

### DISCUSSION THREAD

**Discussion by lexoyo** [GrapesJS/grapesjs#4980] [Ideas]
Hello
Has someone already thought about how to use [CSS functions](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Functions) such as `calc` in the `width` style prop for example?
I don't think we can address this using the [style manager](https://grapesjs.com/docs/getting-started.html#style-manager) (or can we?)
I would be willing to contribute this feature if it is needed, but what should the UX be exactly?

**Comment by artf** [comment]
That's an interesting question, currently when you try to update some value of the Style Manager properties, [those values are parsed](https://github.com/GrapesJS/grapesjs/blob/a2ca4b1a93520e61cc7e49a42f103d5887e33cf2/src/style_manager/model/Property.ts#L303) based on the property itself (eg. the number property has to extract units). If there is a need to allow any kind of value on the property we should allow to skip the "value parser" step and on the UI/UX side a simple text input could be used.

**Reply by lexoyo** [reply]
Let me add this to my todo of things to contribute in order to get to know more grapesjs
Thank you for the answer
